### DISCUSSION TITLE
Enhancements and upgrades for the release: v0.2 🚀

### DISCUSSION THREAD

**Discussion by Mayankm96** [isaac-sim/IsaacLab#106] [Announcements]
Since the unveiling of `v0.1`, we have been diligently working to enhance the functionality of our framework. Through our collective efforts, we are focused on refining the framework to offer a more comprehensive and modular solution. Here's an overview of the forthcoming changes:

* **Improved asset management:** New APIs for loading assets directly from different file types (such as URDF and USD)
* **Code refactoring:** Necessary simplification of the interfaces to prevent code duplication where possible and allow reusability between different workflows
* **Expanded manager functions:** More managers to handle/configure terminations, curriculum, randomization, and action spaces. These additions will enhance the environment designing experience.
* **Sensor updates:** New sensor classes (such as camera, contact sensors, custom ray-casters) that allow obtaining sensor data in a batched format with a notion of lazy-updates to obtain data at different frequencies
* **Procedural terrain generation:** Support for generating different terrains not only from [height fields](https://github.com/NVIDIA-Omniverse/OmniIsaacGymEnvs/blob/main/omniisaacgymenvs/tasks/utils/anymal_terrain_generator.py) but also using the [trimesh](https://trimsh.org/trimesh.html) library
* **More creative scene designing:** Support for non-homogenous environments that allow instances to differ in terms of spawned assets. For example, having different objects for grasping on different tables

We understand that these updates will result in changes to existing public APIs. While this might require some adjustments, our aim is to offer improved functionality and a smoother experience. We'll provide guidance for migrating to the new interfaces.

Please note that these changes have led to deviations from our previously shared [roadmap](https://isaac-orbit.github.io/orbit/source/refs/roadmap.html).  As we integrate features that rely on the upcoming Isaac Sim 2023.1 release, our goal is to align this release with the upcoming public launch of Isaac Sim.

Thank you for your ongoing support! Please feel free to use this forum for further discussions.

**Comment by Yinghui-Lee** [comment]
When will v1.0 be released?

**Reply by Yinghui-Lee** [reply]
Happily Isaac sim 2023.1.0 is released now. when does the orbit repo update?

**Comment by adamwhats** [comment]
Great news!

**Comment by zhuoqun-chen** [comment]
Also can't wait for the Isaac Sim new release. It's gonna upgrade from python `3.7` to `3.10` right? Can't wait!

**Comment by BrightAlex** [comment]
Would 6 6-axis force/torque sensors be supported in the future?

**Comment by songlin** [comment]
Hello, When will the new version be released? It's mid-octorber now :)

**Reply by b-vm** [reply]
Also want to know this

**Comment by Mayankm96** [comment]
Hi everyone,

Thank you for pointing out the [Isaac Sim 2023.1 release](https://forums.developer.nvidia.com/t/isaac-sim-2023-1-0-release-announcement/269863). It has many exciting new features and fixes that we are looking forward to trying out.

Currently, we are checking that the Orbit interfaces and environments are still compatible and there are no breaking changes or roadblocks introduced from this upgrade. We will keep you all updated on our progress.

**Comment by catachiii** [comment]
Hi Mayank, 

Happy to see all these updates! 

May I ask whether you have decided to make the whole Orbit Python 3.10 compatible? + Replacing the old gym libs with the newer gymnasium?

Cannot wait to see the new release ;)

**Reply by Mayankm96** [reply]
Hi @catachiii ,

Yes, we have the framework compatible with Python 3.7 (Isaac Sim 2022.2.1) and Python 3.10 (Isaac SIm 2023.1). However, we suggest all new users to use 2023.1.0-hotfix.1 release since it has some more fixes that came in recently.

Additionally, with the latest changes, we have now migrated the environments to support Gymnasium 0.29.0! So everything should be now with the latest dependencies :)

**Comment by Yinghui-Lee** [comment]
When can we use the v0.2 release?

**Reply by Guanbin-Huang** [reply]
When can we use the v0.2 release. Desire for that too!

**Comment by Abi-64** [comment]
Any updates guys? :)

**Comment by Mayankm96** [comment]
We extend our sincere apologies for the delay in bringing you the awaited release. The process of aligning documentation and tutorials took more time than initially anticipated.

We are excited to announce that the development branch has now been successfully merged into the main branch. This marks a significant leap forward in our framework's evolution.

Please explore the latest changes and updates on our main branch. We are confident that these advancements will elevate your work with our framework.
